src/state/component-repository/hub/actions.js

```diff
@@ -260,17 +261,24 @@ export const sendDeployBundle = bundle => (dispatch, getState) => (
   })
 );
 
-export const sendUndeployBundle = bundle => dispatch => (
+export const sendUndeployBundle = bundle => (dispatch, getState) => (
   new Promise((resolve) => {
     dispatch(toggleLoading(`undeployBundle${bundle.gitRepoAddress}`));
-    undeployBundle(bundle).then((response) => {
+    undeployBundle(bundle.componentCode).then((response) => {
       response.json().then((data) => {
         if (response.ok) {
           dispatch(addToast(
             { id: 'app.undeployed', values: { type: 'bundle', code: bundle.name } },
             TOAST_SUCCESS,
           ));
-          // @TODO do same as above when deploying is done
+          dispatch(fetchSelectedBundleStatus(bundle.gitRepoAddress || bundle.componentUrl));
+          const state = getState();
+          const components = getECRComponentList(state);
+          dispatch(setECRComponents([...components.filter(c => c.code !== bundle.componentCode)]));
```

Comment:
not sure why you still need to array spread this part. seems redundant

Reply:
ah right. let me change it